python/lsst/daf/butler/direct_butler/_direct_butler.py

```diff
@@ -1808,22 +1809,58 @@ def ingest(
 
         import_info = self._prepare_ingest_file_datasets(datasets, progress)
 
+        new_datasets, existing_datasets = self._partition_datasets_by_known(datasets)
```

Comment:
Maybe move this chunk before the `_prepare_ingest_file_datasets` so we can fail quicker?